ci/ci-test-tasks/main.yml

```diff
@@ -6,78 +6,103 @@ parameters:
 - name: BuildSourceVersion
   displayName: The latest version control change of the triggering repo that is included in this build.
   type: string
+- name: PushNugetPackage
```

Comment:
We should remove the parameters PushNugetPackage and PublishHotfixArtifact at all since they always have the same value. The original sources shows: 
 parameters:
            task: ${{ task }}
            push: false
            publishArtifact: true
            patchVersion: true
            version: '0.999.0'
            BuildSourceVersion: ${{ parameters.BuildSourceVersion }}
         
So I suggest just get rid of the push, publishArtifact and patchVersion parameters and any corresponding code that depends on it. 
It seems it was copied from another pipeline definition where those parameters are actually used.

Reply:
These steps are not always needed that's why I introduced flags for them. In most cases nuget pushing and hotfix artifact uploading are not required.

Reply:
I think I didn't comment on some parts clearly. What I mean is we do not need them at all since they are not changing. It is constant:
push: false
publishArtifact: true	
patchVersion: true

According to this we can get rid of them at all.

Reply:
Most of this pipeline runs are to test new PRs, for such case we don't need to push nuget packages and publish artifacts. We only need to patch version of the task and test it with ci tests. So configuration of flags will be like this:
push: false, publishArtifact: false, patchVersion: true
When we need a hotfix we need the following configuration of flags:
push: true, publishArtifact: true, patchVersion: false
For a rare case when we don't need a hotfix but only a nuget package for some reason flags will be the following:
push: true, publishArtifact: false, patchVersion: false
So as you can see we do need PushNugetPackage and PublishHotfix flags since there multiple permutations of those.
patchVersion might be resolve as not(push) though

Reply:
After reading other comments I've realized that we don't need hotfix or nuget push functionality here. So I will remove these flags then.